proto/chan_config.go

```diff
@@ -0,0 +1,518 @@
+	SignaturePolicy    *common.SignaturePolicyEnvelope `json:"signature,omitempty"`
+}
+
+// Certificate - describes certificate(can be ca, intermediate, admin) from msp
+type Certificate struct {
+	FingerprintSHA256 []byte
+	Data              []byte
+	MSPID             string
+	Type              CertType
+	MSPName           string
+}
+
+type CertType string
+
+const (
+	RootCACertType       CertType = "ca"
+	IntermediateCertType CertType = "intermediate"
+	AdminCertType        CertType = "admin"
+)
+
+func ParseChannelConfig(cc common.Config) (*ChannelConfig, error) {
+	chanCfg := &ChannelConfig{}
+
+	appCfg, err := ParseApplicationConfig(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse application config: %w", err)
+	}
+	chanCfg.Applications = appCfg
+
+	orderers, err := ParseOrderer(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse orderers config: %w", err)
+	}
+	chanCfg.Orderers = orderers
+
+	batchSize, err := ParseOrdererBatchSize(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse batch size: %w", err)
+	}
+	chanCfg.OrdererBatchSize = *batchSize
+
+	batchTimeout, err := ParseOrdererBatchTimeout(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse batch timeout: %w", err)
+	}
+	chanCfg.OrdererBatchTimeout = batchTimeout
+
+	consensusType, err := ParseOrdererConsesusType(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse consensus type: %w", err)
+	}
+	chanCfg.OrdererConsesusType = *consensusType
+
+	consortium, err := ParseConsortium(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse consortium: %w", err)
+	}
+	chanCfg.Consortium = consortium
+
+	hashingAlgorithm, err := ParseHashingAlgorithm(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse hashing algorithm: %w", err)
+	}
+	chanCfg.HashingAlgorithm = hashingAlgorithm
+
+	blockDataHashing, err := ParseBlockDataHashingStructure(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse block data hashing structure: %w", err)
+	}
+	chanCfg.BlockDataHashingStructure = *blockDataHashing
+
+	capabilities, err := ParseCapabilities(cc)
+	if err != nil {
+		return nil, fmt.Errorf("parse capabilities: %w", err)
+	}
+	chanCfg.Capabilities = *capabilities
+
+	policies, err := ParsePolicy(cc.ChannelGroup.Policies)
+	if err != nil {
+		return nil, fmt.Errorf("parse policies: %w", err)
+	}
+
+	chanCfg.Policy = policies
+
+	return chanCfg, nil
+}
+
+func ParseApplicationConfig(cfg common.Config) (map[string]ApplicationConfig, error) {
+	applicationGroup, exists := cfg.ChannelGroup.Groups[channelconfig.ApplicationGroupKey]
+	if !exists {
+		return nil, fmt.Errorf("application group doesn't exists")
+	}
+
+	appCfg := map[string]ApplicationConfig{}
+
+	for groupName := range applicationGroup.Groups {
+		mspCfg, err := ParseMSP(applicationGroup.Groups[groupName])
+		if err != nil {
+			return nil, fmt.Errorf("parse msp: %w", err)
+		}
+
+		ancPeers, err := ParseAnchorPeers(applicationGroup.Groups[groupName].Values[channelconfig.AnchorPeersKey].Value)
```

Comment:
[]byte attr too common. Suggest to pass applicationGroup.Groups[groupName] as parameter